api_tests/e2e/rfc003/btc_eth/inverted-refund.ts

```diff
@@ -0,0 +1,139 @@
+import * as bitcoin from "../../../lib/bitcoin";
+import { Actor } from "../../../lib/actor";
+import { ActionKind, SwapRequest } from "../../../lib/comit";
+import { toBN, toWei } from "web3-utils";
+import { HarnessGlobal } from "../../../lib/util";
+import { Step, createTests } from "../../test_creator";
+import { expect } from "chai";
+import "chai/register-should";
+import "../../../lib/setupChai";
+
+declare var global: HarnessGlobal;
+
+(async function() {
+    const alice = new Actor("alice", global.config, global.project_root, {
+        ethereumNodeConfig: global.ledgers_config.ethereum,
+        bitcoinNodeConfig: global.ledgers_config.bitcoin,
+        addressForIncomingBitcoinPayments:
+            "bcrt1qs2aderg3whgu0m8uadn6dwxjf7j3wx97kk2qqtrum89pmfcxknhsf89pj0",
+    });
+    const bob = new Actor("bob", global.config, global.project_root, {
+        ethereumNodeConfig: global.ledgers_config.ethereum,
+        bitcoinNodeConfig: global.ledgers_config.bitcoin,
+        addressForIncomingBitcoinPayments: null,
+    });
+
+    const alphaAssetQuantity = 100000000;
+    const betaAssetQuantity = toBN(toWei("10", "ether"));
+
+    const alphaExpiry = Math.round(Date.now() / 1000) + 13;
+    const betaExpiry = Math.round(Date.now() / 1000) + 8;
+
+    await bitcoin.ensureFunding();
+    await bob.wallet.eth().fund("11");
+    await alice.wallet.eth().fund("0.1");
+    await alice.wallet.btc().fund(10);
+    await bitcoin.generate();
+
+    let swapRequest: SwapRequest = {
+        alpha_ledger: {
+            name: "bitcoin",
+            network: "regtest",
+        },
+        beta_ledger: {
+            name: "ethereum",
+            network: "regtest",
+        },
+        alpha_asset: {
+            name: "bitcoin",
+            quantity: alphaAssetQuantity.toString(),
+        },
+        beta_asset: {
+            name: "ether",
+            quantity: betaAssetQuantity.toString(),
+        },
+        beta_ledger_redeem_identity: alice.wallet.eth().address(),
+        alpha_expiry: alphaExpiry,
+        beta_expiry: betaExpiry,
+        peer: await bob.peerId(),
+    };
+
+    const steps: Step[] = [
+        {
+            actor: bob,
+            action: ActionKind.Accept,
+            waitUntil: state => state.communication.status === "ACCEPTED",
+        },
+        {
+            actor: alice,
+            action: ActionKind.Fund,
+            waitUntil: state => state.alpha_ledger.status === "Funded",
+        },
+        {
+            actor: bob,
+            action: ActionKind.Fund,
+            waitUntil: state =>
+                state.alpha_ledger.status === "Funded" &&
+                state.beta_ledger.status === "Funded",
+        },
+        {
+            actor: alice,
+            action: ActionKind.Refund,
+            waitUntil: state => state.alpha_ledger.status === "Refunded",
+        },
+        {
```

Comment:
We have a possible race-condition here that could lead to flacky tests.

In short, it is not guaranteed that Bob's comit node sees the refund at the same time as Alice's node does, however, the last action only does `waitUntil` for Alice.

I think this step should be changed to mimic the step above, e.g.:

```js
{
   actor: bob,
   waitUntil: state => state.alpha_ledger.status === "Refunded",
},
```

Reply:
Yes I agree, thanks.

Reply:
Implemented as suggested.

Reply:
> Implemented as suggested.

Where?
The `step` I made this comment on still asserts that `bob` sees the `alpha_ledger.status` as `Refunded` even though he doesn't wait for it.